example-application/test/basic-tests.test.js

```diff
@@ -120,7 +142,25 @@ describe("/api", () => {
       expect(scope.isDone()).toBe(true);
     });
 
-    test("When the user does not exist, return http 404", async () => {
+    test("When adding an order without specifying product, stop and return 400", async () => {
+      //Arrange
+      nock("http://localhost/user/").get(`/1`).reply(200, {
+        id: 1,
+        name: "John",
+      });
+      const orderToAdd = {
+        userId: 1,
+        mode: "draft",
+      };
+
+      //Act
+      const orderAddResult = await request(expressApp).post("/order").send(orderToAdd);
+
+      //Assert
+      expect(orderAddResult.status).toBe(400);
```

Comment:
We don't stand to the test title promise - We only test the response. Shouldn't we check that it's not retrievable?

Reply:
How do we check it if we don't receive any id from the post request?

Reply:
There is usually a way to retrieve a record, some field can be set with unique value. For example, IF order had a field called order.remarks we could do order.remarks = something-unique, query for it and ensure nothing returns. 

In our case, we have too few fields, I would consider enriching the payload a bit, like adding "remarks", "totalToPay"

p.s. we didn't fill the field "productId"